### PR TITLE
feat(SSO login) : Support SSO using OAuth 2.0 ✨ 

### PR DESCRIPTION
**Pull request by Athira001**
https://github.com/quintype/ace-planning/milestone/22
https://docs.google.com/document/d/1DkYSdOaIxICuuwIBz197WQqQedO1hxmU2WimPPyJ49Q/edit
- Support user login, Register, Forgot password, social login etc

- Hide header and footer from user-login page

image

https://github.com/quintype/bridgekeeper-js/pull/18/files

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test Case A
- [ ] Test Case B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: config/publisher.yml [hunk 1]
```diff
@@ -6,9 +6,17 @@ host_to_automatic_api_host:
...
 domain_mapping:
-  subdomain.lvh.me: subdomain
+  subdomain.lvh.me: voices
   malibu.lvh.me: localhost:3000
 publisher:
+  sso_login:
+    is_enable: true
+    redirect_Url: "http://malibu.lvh.me:3000/api/auth/v1/oauth/token"
+    callback_Url: "https://malibu-advanced-web.qtstage.io"
+    client_id : 51
```
Comment (Athira001) [inline config/publisher.yml:16]: We will remove it from here and will keep it in BK

File: app/isomorphic/components/layouts/header/nav-bar/index.js [hunk 2]
```diff
@@ -19,12 +19,20 @@ const NavBar = () => {
...
+  const enableLogin = get(publisherAttributes, ["enableLogin"], true);
   const isHamburgerMenuOpen = useSelector(state => get(state, ["isHamburgerMenuOpen"], false));
   const menu = useSelector(state => get(state, ["qt", "data", "navigationMenu", "homeMenu"], []));
   const hamburgerMenu = useSelector(state => get(state, ["qt", "data", "navigationMenu", "hamburgerMenu"], []));
-
   const displayStyle = isHamburgerMenuOpen ? "flex" : "none";
+  const domainSlug = useSelector(state => get(state, ["qt", "config", "domainSlug"], ""));
+  const clientId = get(publisherAttributes, ["sso_login", "client_id"], "");
+  const redirectUrl = domainSlug
+    ? get(publisherAttributes, ["sso_login", "subdomain", "redirect_Url"], "")
```
Comment (ags1773) [inline app/isomorphic/components/layouts/header/nav-bar/index.js:31]: what if there are multiple subdomains?

Comment (Athira001) [reply app/isomorphic/components/layouts/header/nav-bar/index.js:31]: In the same way, we need to pass different redirect URLs for a different domain.  Also `redirect_uri` needs to be whitelisted in Brke.

Comment (ags1773) [reply app/isomorphic/components/layouts/header/nav-bar/index.js:31]: How will the config look like for two or more subdomains?

Comment (Athira001) [reply app/isomorphic/components/layouts/header/nav-bar/index.js:31]:
```
    subdomain:
      voices:
        redirect_Url: "https://malibu-voices-advanced-web.qtstage.io/api/auth/v1/oauth/token"
        callback_Url: "https://malibu-voices-advanced-web.qtstage.io"
      hindi:   
        redirect_Url: "https://malibu-voices-advanced-web.qtstage.io/api/auth/v1/oauth/token"
        callback_Url: "https://malibu-voices-advanced-web.qtstage.io"
      epaper:
        redirect_Url: "https://malibu-voices-advanced-web.qtstage.io/api/auth/v1/oauth/token"
        callback_Url: "https://malibu-voices-advanced-web.qtstage.io"
```
let  redirectUrl =""
if (domainSlug === "voices") return redirectUrl = get(publisherAttributes, ["sso_login", "subdomain", "voices", "redirect_Url"], "")
else if (domainSlug === "hindi") return redirectUrl = get(publisherAttributes, ["sso_login", "subdomain", "hindi", "redirect_Url"], "")
else if (domainSlug === "epaper")return redirectUrl = get(publisherAttributes, ["sso_login", "subdomain", "epaper", "redirect_Url"], "")
else return redirectUrl = get(publisherAttributes, ["sso_login", "redirect_Url"], "")

Or we can use `switch `

File: app/isomorphic/components/layouts/header/nav-bar/index.js [hunk 3]
```diff
@@ -19,12 +19,21 @@ const NavBar = () => {
...
   const menu = useSelector(state => get(state, ["qt", "data", "navigationMenu", "homeMenu"], []));
   const hamburgerMenu = useSelector(state => get(state, ["qt", "data", "navigationMenu", "hamburgerMenu"], []));
-
   const displayStyle = isHamburgerMenuOpen ? "flex" : "none";
+  const domainSlug = useSelector(state => get(state, ["qt", "config", "domainSlug"], ""));
+  const clientId = get(publisherAttributes, ["sso_login", "client_id"], "");
+  const redirectUrl =
+    domainSlug === "voices"
+      ? get(publisherAttributes, ["sso_login", "subdomain", "voices", "redirect_Url"], "")
+      : get(publisherAttributes, ["sso_login", "redirect_Url"], "");
```
Comment (ags1773) [inline app/isomorphic/components/layouts/header/nav-bar/index.js:33]: why are you hardcoding `domainSlug`?

**Review by ags1773 — APPROVED** [review]

**Review by shraddha-kesari — APPROVED** [review]

**Review by Jeevan-Kishore — APPROVED** [review]

**Pull request merged**